### PR TITLE
fix: load subitems on search

### PR DESCRIPTION
**Pull request by cbourgois**
### Description of the Change

* add `forceDisplaySearch` to force the search input display
* fix bug when using `onLoad` and search
* force to load subitems on search

e80de20 — fix: load subitems on search

ref: DTRUN-2021

### CODE REVIEW

File: src/list/item/factory.js [hunk 1]
```diff
@@ -535,15 +536,29 @@ export default /* @ngInject */ function ($q, $timeout) {
...
           }
-        });
-        self.displaySearchResults(filteredItems);
+          return results;
+        }, []);
+
+        $q.all(promises)
+          .then(() => {
+            let tmpMatchingItem;
+            angular.forEach(self.subItemsAdded, (item) => {
```
Comment (FredericEspiau) [inline src/list/item/factory.js]:
```suggestion
            const filteredItems = _.filter( // not sure filteredItems isn't used elsewhere
              _.map(
                self.subItemsAdded,
                item => getMatchingItem(item, search)
              ),
            );
```

File: src/list/item/factory.js [hunk 2]
```diff
@@ -535,15 +536,29 @@ export default /* @ngInject */ function ($q, $timeout) {
...
         search = search.toLowerCase(); // ignore case
         const filteredItems = [];
-        let tmpMatchingItem;
-        angular.forEach(self.subItemsAdded, (item) => {
-          tmpMatchingItem = getMatchingItem(item, search);
 
-          if (tmpMatchingItem) {
-            filteredItems.push(tmpMatchingItem);
+        const promises = _.reduce(self.subItemsAdded, (results, item) => {
+          if (item.onLoad && !item.isLoaded) {
```
Comment (marie-j) [inline src/list/item/factory.js]: This looks like a filter + map no ? 
```suggestion
         filter(self.subItemsAdded, item => item.onLoad && !item.isLoaded).map(item => item.loadSubItems())
```
I don't know if that's better in terms of readability/performances

Comment (FredericEspiau) [reply src/list/item/factory.js]: I said the exact same thing yesterday but my comment isn't the for some reason !

I agree fully with you :)

**Comment by antleblanc** [comment]
@cbourgois Could you please pay attention to the build pipeline which is reporting a failure?

Thanks!

**Pull request merged**